src/EStimLibrary/Core/ResourceManager.cs

```diff
@@ -29,7 +31,7 @@ public bool IsValidResourceId(int globalId)
     {
         // TODO: does Used check validity again?
         return this.IdPool.IsValidId(globalId) &&
-            this.IdPool.IsUsed(globalId);
+            this.IdPool.IsUsed(globalId) && this.Resources.ContainsKey(globalId);
```

Comment:
Is this line needed? Was it failing before? the IdPool.IsValidID + IsUsed checks should be equivalent, so if it was failing before, something bigger is broken

Reply:
I had a misunderstanding of the code. I mistook the id being used will make it invalid while in contrast it actually verifies the validity. I used Assert.False instead of True. I have fixed the test case

Reply:
For the current implementation of IsValidId and IsUsed, it is crucial to have the line since we are checking for the resource. I have tried to implement the check in UseId but i can't do that without initiating an instance of ResourceManager, so I think it's simpler to have the check inside the RMgr class.